### PR TITLE
find TLE for given cartesian state vector using Nelder-Mead optimization

### PR DESCRIPTION
**Pull request by spire-josh-singer**
Idea is to transform specific cartesian state vector to a TLE such that if SGP4 is initialized with those parameters, it would produce the state vector if propagated to that epoch.

the method is

- transform the cartesian state vector to purely keplerian orbit parameters
- Use these as a starting point for an optimization method:
- https://en.wikipedia.org/wiki/Nelder%E2%80%93Mead_method

We vary all 6 orbital parameters in the TLE. The objective function runs SGP4 and figures out the sum of squares error in the position and velocity vectors.

### CODE REVIEW

File: src/tlegen.rs [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+            self.coe.argument_of_perigee.get::<degree>(),
+            self.coe.mean_anomaly.get::<degree>(),
+            self.coe.semimajor_axis.get::<kilometer>(),
+        ];
+        let mut initial_simplex: Vec<Vec<f64>> = vec![];
+        for _i in 0..7 {
+            // initial simplex requires n+1 vertices
+            initial_simplex.push(init_param.clone());
+        }
+        let perturbations = vec![0.1, 0.1, 0.01, 1.0, 5.0, 1.0];
```
Comment (elliemcc-spire) [inline src/tlegen.rs:76]: Little clippy warning here for vec

**Review by elliemcc-spire — CHANGES_REQUESTED** [review]
Looking good - just a small clippy warning for vec

**Review by elliemcc-spire — APPROVED** [review]
LGTM

File: src/tlegen.rs (hunk 1 above)
Comment (spire-josh-singer) [reply src/tlegen.rs]: fixed

**Pull request merged**